### ISSUE TITLE
False positive Lint/EmptyWhen offense when using endless ranges in a case statement

### ISSUE THREAD

**Issue by stillhart** [rubocop/rubocop#10993]
Rubocop shows me a `Lint/EmptyWhen` offense when I create a `case` statement with an [endless range](https://rubyreferences.github.io/rubychanges/2.6.html#endless-range-1). I think this is a bug. The offense does not show up when I don't use an endless range.

--------

## Expected behavior

Rubocop shows no offense when using endless ranges in case statements.

## Actual behavior

Rubocop show's an offense despite the non-empty body in the when statement.
```
Lint-EmptyWhen.rb:3:1: W: Lint/EmptyWhen: Avoid when branches without a body.
when 1.. ...
^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Create a file named `Lint-EmptyWhen.rb` with the following content
```ruby
case 1
when 1..
  puts("this is a body and it is present")
end
```

Run the Lint/EmptyWhen cop:
```
bundle exec rubocop Lint-EmptyWhen.rb --only "Lint/EmptyWhen"
```
Rubocop will show the `Lint/EmptyWhen` offense.

## RuboCop version

```
bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.0.4) [x86_64-linux]
```

**Comment by koic** [comment]
This is not a false positive because it is the same as:

```ruby
case 1
when 1..puts("this is a body and it is present")
end
```

Presumably what is expected is the need for parentheses.

```ruby
case 1
when (1..)
  puts("this is a body and it is present")
end
```

**Comment by stillhart** [comment]
🤯 Oh wow, I didn't see that coming. Thank you very much for the fast and perfect response.
